IntroOOP/RatNum.cs

```diff
@@ -0,0 +1,369 @@
+        RatNum a = this.Reduce();
+        RatNum b = x.Reduce();
+        return a.Numerator == b.Numerator && a.Denominator == b.Denominator;
+    }
+    // Метод сравнения двух дробей
+    // 0, если дроби равны
+    // 1, если А дробь больше В
+    // -1, если А меньше В
+    private int CompareTo(RatNum x)
+    {
+        if (this.Equals(x))
+        {
+            return 0;
+        }
+        RatNum a = this.Reduce();
+        RatNum b = x.Reduce();
+        if (a.Numerator / a.Denominator > b.Numerator / b.Denominator)
+            return 1;
+        else return -1;
+    }
+    // Перегрузка оператора "+". Сложение двух дробей.
+    public static RatNum operator +(RatNum a, RatNum b)
+    {
+        return PerformOperationAddition(a, b);
+    }
+    // Перегрузка оператора "+". Сложение дроби с числом.
+    public static RatNum operator +(RatNum a, int b)
+    {
+        return a + new RatNum(b);
+    }
+    // Перегрузка оператора "+"Сложение. числа с дробью.
+    public static RatNum operator +(int a, RatNum b)
+    {
+        return b + a;
+    }
+    // Перегрузка оператора "-". Вычитание двух дробей.
+    public static RatNum operator -(RatNum a, RatNum b)
+    {
+        return PerformOperationSubtraction(a, b);
+    }
+    // Перегрузка оператора "-".Вычитание из дроби числа.
+    public static RatNum operator -(RatNum a, int b)
+    {
+        return a - new RatNum(b);
+    }
+    // Перегрузка оператора "-".Вычитание из числа дроби.
+    public static RatNum operator -(int a, RatNum b)
+    {
+        return b - a;
+    }
+
+    // Перегрузка оператора "*". Умножение дробей.
+    public static RatNum operator *(RatNum a, RatNum b)
+    {
+        return new RatNum(a.Numerator * b.Numerator, a.Denominator * b.Denominator);
+    }
+
+    // Перегрузка оператора "*". Произведение дроби и числа.
+    public static RatNum operator *(RatNum a, int b)
+    {
+        return a * new RatNum(b);
+    }
+    // Перегрузка оператора "*". Произведение числа и дроби.
+    public static RatNum operator *(int a, RatNum b)
+    {
+        return b * a;
+    }
+
+    // Перегрузка оператора "/". Деление двух дробей.
+    public static RatNum operator /(RatNum a, RatNum b)
+    {
+        return a * b.ReverseNum();
+    }
+
+    // Перегрузка оператора "/". Деление дроби на число.
+    public static RatNum operator /(RatNum a, int b)
+    {
+        return a / new RatNum(b);
+    }
+    // Перегрузка оператора "/". Деление числа на дробь.
+    public static RatNum operator /(int a, RatNum b)
+    {
+        return new RatNum(a) / b;
+    }
+    // Перегрузка оператора "++"
+    public static RatNum operator ++(RatNum a)
+    {
+        return a + 1;
+    }
+    // Перегрузка оператора "--"
+    public static RatNum operator --(RatNum a)
+    {
+        return a - 1;
+    }
+
+    // Перегрузка оператора "==". Сравнение дробей с помощью созданого метода.
+    public static bool operator ==(RatNum a, RatNum b)
+    {
+        return a.Equals(b);
```

Comment:
Если дробь в вашем исполнении - это класс, то значением `a` вполне может быть `null`, а в этом случае ваша программа "ляжет" при использовании этого оператора. Надо делать так:
```suggestion
        return Equals(a, b);
```